### PR TITLE
Fix stable sorting for emoji-only thoughts

### LINKED ISSUE #4847: Stable sort lone emoji like empty thoughts
## Steps to Reproduce

```
- X
  - =sort
    - Alphabetical
  - A
  - B
  - C
```

1. Set the cursor on `X`. 
2. New Subthought (Cmd + Enter).
3. Type 🙂. 
4. Type additional characters. 

## Current Behavior

🙂 is sorted to the top immediately (step 3). 

image

## Expected Behavior

🙂 should not be sorted to the top (step 3). 

image

🙂 should be sorted to the top only after additional characters are typed (step 4), just like empty thoughts. 

Thoughts that consist only of emoji should have the same sort behavior as empty thoughts. 

## Notes

The above steps only show one of several cases that involve stable sort behavior, but you should not need to enumerate or test every one if the empty thought condition is simply expanded to include lone emoji.

Unlike the empty thought, the emoji is typed, which triggers onChange. That might require different handling. However, empty thoughts maintain a stable sort position not just when they are create through New Subthought, but also when they are created through deleting characters in an existing thought. So I suspect that the machinery is already there.

### PR DESCRIPTION
**Pull request by patbyte**
Issue #4847

## Summary

- Treat thoughts containing only emoji as empty thoughts for stable sorting.
- Preserve their insertion position until non-emoji text is entered.
- Apply the same behavior when creating thoughts, editing thoughts, and validating sort order.
- Preserve the existing emoji-first ordering once a thought contains additional text.

## Implementation

A shared `isEmptyOrEmojiOnly` predicate expands the existing empty-thought condition without changing the alphabetical comparator itself.

Regression coverage verifies that:

1. An empty thought is created at the end of an alphabetically sorted context.
2. Typing `🙂` does not move it.
3. Typing `🙂a` moves it to its normal alphabetical position.

### CODE REVIEW

File: src/util/isEmptyOrEmojiOnly.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { REGEX_EMOJI_GLOBAL } from '../constants'
+import stripTags from './stripTags'
+
+/** Returns true when a thought is empty or its visible content consists only of emoji and whitespace. */
+const isEmptyOrEmojiOnly = (value: string): boolean => {
+  const visibleValue = stripTags(value)
+  return (
+    value === '' ||
+    (!!visibleValue.match(REGEX_EMOJI_GLOBAL) && visibleValue.replace(REGEX_EMOJI_GLOBAL, '').trim() === '')
```
Comment (raineorshine) [inline src/util/isEmptyOrEmojiOnly.ts:9]: This whole return statement can be simplified to `visibleValue.replace(REGEX_EMOJI_GLOBAL, '').trim() === ''`. It has the same effect for empty string and lone emoji, and the added bonus of treating empty HTML tags like `<b></b>` as empty as well.

See 58f6c99f78

**Review by raineorshine — APPROVED** [review]
Thank you

File: src/util/isEmptyOrEmojiOnly.ts (hunk 1 above)
Comment (patbyte) [reply src/util/isEmptyOrEmojiOnly.ts:9]: Thank you, @raineorshine  :+1:
